packages/mui-material/src/TextField/TextField.d.ts

```diff
@@ -270,4 +276,12 @@ export type TextFieldProps = StandardTextFieldProps | FilledTextFieldProps | Out
  * - [TextField API](https://mui.com/material-ui/api/text-field/)
  * - inherits [FormControl API](https://mui.com/material-ui/api/form-control/)
  */
-export default function TextField(props: TextFieldProps): JSX.Element;
+export default function TextField<V extends Variants>(
```

Comment:
Would that be a breaking change where it will be mandatory to provide one generic argument type? Perhaps it should have a default value:
```suggestion
export default function TextField<V extends Variants = Variants>(
```

Reply:
I added a default value [here](https://github.com/mui/material-ui/blob/0d51f44779233d05b1519e69f2bbaac1fbd43e09/packages/mui-material/src/TextField/TextField.d.ts#L231), so it won't be breaking change even if variant is not provided

codesandbox: https://codesandbox.io/s/festive-bohr-g2t069?file=/demo.tsx